src/video/wayland/SDL_waylandevents.c

```diff
@@ -328,8 +337,18 @@ ProcessHitTest(struct SDL_WaylandInput *input, uint32_t serial)
            WL_SHELL_SURFACE_RESIZE_*), but the values are the same. */
         const uint32_t *directions_zxdg = directions_wl;
 
+#ifdef HAVE_LIBDECOR_H
+        /* ditto for libdecor. */
+        const uint32_t *directions_libdecor = directions_wl;
+#endif
+
         switch (rc) {
             case SDL_HITTEST_DRAGGABLE:
+#ifdef HAVE_LIBDECOR_H
+                if (input->display->shell.libdecor) {
```

Comment:
This and the block below need a null check for `window_data->shell_surface.libdecor.frame`, similar to the toplevel null checks.